### PR TITLE
fix(server): improved packfile encoding to support short messages and…

### PR DESCRIPTION
**Pull request by sammoorhouse**
… unicode in errors

This change allows you to say

```
step.setError(`🥺 Rejecting repo ${action.repo} not in the authorisedList`);
```

in a step.

Also fixes a padding bug that meant fewer than 5chars of errors broke the proxy response.

### CODE REVIEW

**Comment by maoo** [comment]
Thanks for the PR @sammoorhouse ! Please reach out to help@finos.org if you encounter any issue to sign the FINOS CLA using EasyCLA.

**Comment by JamieSlome** [comment]
@sammoorhouse - thanks for the contribution 👍 I'd been wondering how to get some "prettier" messages into the CLI for the developer.

Do you think we could include a test case with this PR to demonstrate it works successfully? Perhaps to demonstrate that a message successfully returns with Unicode characters?

Happy to help if needed 🎉

**Comment by sammoorhouse** [comment]
Thanks @JamieSlome I'll have a look later today!

**Comment by sammoorhouse** [comment]
I extracted 'handleMessage' in proxy/routes and wrote two tests just for that function in a second commit just now. Thanks!

File: src/proxy/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable max-len */
 const proxyApp = require('express')();
 const bodyParser = require('body-parser');
-const routes = require('./routes');
+const routes = require('./routes').router;
```
Comment (JamieSlome) [inline src/proxy/index.js:4]: @sammoorhouse, can you explain why this change is needed?

Comment (sammoorhouse) [reply src/proxy/index.js:4]: Because I expose {router, handleMessage} from proxy/routes/index.js

File: src/proxy/routes/index.js [hunk 2]
```diff
@@ -69,4 +64,13 @@ router.use(
   }),
 );
 
-module.exports = router;
+const handleMessage = async(message) => {
+  const errorMessage = `ERR\t${message}`;
+  const len = 6 + new TextEncoder().encode(errorMessage).length;
+
+  const prefix = len.toString(16);
+  const packetMessage = `${prefix.padStart(4, '0')}\x02${errorMessage}\n0000`;
```
Comment (psmulovics) [inline src/proxy/routes/index.js:72]: Nice!

**Review by psmulovics — APPROVED** [review]

**Comment by psmulovics** [comment]
Can someone kick the build again? It failed on something unrelated imho.

**Comment by coopernetes** [comment]
> Can someone kick the build again? It failed on something unrelated imho.

The failure is due to #213, we have some sort of repo permission bug in the coverage steps. 

I’ll try to get a PR open soon to get it fixed then we can rebase on this one. Seems like a simple enough fix: https://github.com/zgosalvez/github-actions-report-lcov/issues/150

**Review by JamieSlome — APPROVED** [review]
LGTM! 🍰 

Ready to go once we've addressed the code coverage permissions issue...

(cc) @coopernetes @CyberCitizen01
